doc/sphinxext/redirect_from.py

```diff
@@ -73,28 +73,29 @@ def redirects(self):
         """The mapping of the redirects."""
         return self.data.setdefault('redirects', {})
 
-    def clear_doc(self, docnames):
-        self.redirects.clear()
+    def clear_doc(self, docname):
+        self.redirects.pop(docname, None)
 
     def merge_domaindata(self, docnames, otherdata):
-        for src, dst in otherdata['redirects'].items():
-            if src not in self.redirects:
-                self.redirects[src] = dst
-            elif self.redirects[src] != dst:
+        for docname in docnames:
+            otherdoc = otherdata['redirects'][docname]
+            if self.redirects[docname]:
+                src, dst = self.redirects[docname]
+                _, otherdst = otherdoc
```

Comment:
Can you please give the motivation for this change? I find this quite confusing, but I belive this is wrong. AFAICS from l.194 (`domain.redirects[redirected_reldoc] = current_doc`) `self.redirects` is a mapping "old (non-existing) page" --> "new (existing) page". OTOH the parameter `docnames` contains the existing pages. So, I belive it is incorrect to use it as a key in `self.redirects`.

Reply:
Hi Tim, the original motivation for the change was to match the implementation here: 
https://github.com/sphinx-doc/sphinx/blob/5523c9bbe4d1415777669330411d4f00ad802f18/sphinx/ext/todo.py#L90-L92

I did make a mistake when adding the sanity check for this function, so I will undo the changes and leave it as is.